src/NativeBase/Containers/ListContainer.tsx

```diff
@@ -269,77 +318,143 @@ const ListContainer = (props: {
-        {/* TODO: reinstate when functionality is ready */}
-        {/* <SegmentedPicker options={projectListOptions} /> */}
-
-        {params?.type && listItemsToShow ? (
-          <>
-            {/* Past / Upcoming / My Events choice */}
-            {params.type === ListType.Events && (
-              <EventOptions selected={eventsSelectedOption} />
-            )}
-
-            {/* Quick search for upcoming events (Today / This week / This month) */}
-            {params.type === ListType.Events &&
-              eventsShowUpcomingQuickSearch &&
-              eventsQuickSearchUpcomingChoice && (
-                <SearchResultsView>
-                  <EventSearchUpcomingQuickSearch
-                    selectedButton={eventsQuickSearchUpcomingChoice}
+      /> */}
+      {/* TODO: reinstate when functionality is ready */}
+      {/* <SegmentedPicker options={projectListOptions} /> */}
+
+      <VStack space={2} alignItems="stretch">
+        <VStack mt="4" px="4">
+          <Heading size="xl" marginBottom="15px">
+            Roles
+          </Heading>
+          <HStack space="2" alignItems="center" width="100%">
+            <View flex={1}>
+              <FreeSearchBar
+                handleSubmit={handleRoleSearchChange}
+                handleChangeText={handleRoleSearchChange}
+                handleClearSearch={handleClearSearch}
+                marginBottom="2"
+              />
+            </View>
+            <View
+              marginBottom="2"
+              height="48px"
+              width="48px"
+              bg="#F0F0F0"
+              borderRadius="10px"
+              padding="2"
+              mx="10px"
+              justifyContent="center"
+              alignItems="center"
+            >
+              <IconButton
+                icon={
+                  <Icon
+                    as={MaterialIcons}
+                    name="tune"
+                    size="24px"
+                    color="gray.500"
                   />
-                </SearchResultsView>
-              )}
+                }
+              />
+            </View>
+          </HStack>
+
+          {/* Display filtered roles based on the search */}
+          <SearchResultsView>
+            {roleSearchText && filteredRoles.length > 0 && (
+              <>
+                <SearchResultsLabel>
+                  Roles matching "{roleSearchText}"
+                </SearchResultsLabel>
+                {filteredRoles.map((roleGroup, index) => (
+                  <Text key={index} marginY="0.5">
+                    {roleGroup.groupName}
+                  </Text>
+                ))}
+              </>
+            )}
+            {roleSearchText && filteredRoles.length === 0 && (
+              <Text>No roles found</Text>
+            )}
+          </SearchResultsView>
+        </VStack>
+
+        <VStack>
+          <ScrollView horizontal showsHorizontalScrollIndicator={false}>
+            <ProjectSearchContainer />
```

Comment:
we should rethink about the structure and/or the naming of the components. In this component we put together the freesearchbar with it's extra icon on the right "unwrapped" but then wrapp the tagbuttons with it's choicelists in "ProjectSearchContainer" which is actually not the correct naming as it's actually filtering.